src/platform/plugins/private/presentation_panel/public/panel_actions/show_config_panel_action/show_config_panel_action.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import {
+  EmbeddableApiContext,
+  CanAccessViewMode,
+  apiCanAccessViewMode,
+  getInheritedViewMode,
+  getViewModeSubject,
+  HasReadOnlyCapabilities,
+  hasReadOnlyCapabilities,
+} from '@kbn/presentation-publishing';
+import {
+  Action,
+  FrequentCompatibilityChangeAction,
+  IncompatibleActionError,
+} from '@kbn/ui-actions-plugin/public';
+import { ACTION_SHOW_CONFIG_PANEL } from './constants';
+
+export type ShowConfigPanelActionApi = CanAccessViewMode & HasReadOnlyCapabilities;
+
+const isApiCompatible = (api: unknown | null): api is ShowConfigPanelActionApi => {
+  return hasReadOnlyCapabilities(api) && apiCanAccessViewMode(api);
+};
+
+export class ShowConfigPanelAction
+  implements Action<EmbeddableApiContext>, FrequentCompatibilityChangeAction<EmbeddableApiContext>
+{
+  public readonly type = ACTION_SHOW_CONFIG_PANEL;
+  public readonly id = ACTION_SHOW_CONFIG_PANEL;
+  public order = 50;
+
+  constructor() {}
+
+  public getDisplayName({ embeddable }: EmbeddableApiContext) {
+    if (!isApiCompatible(embeddable)) throw new IncompatibleActionError();
+    // check permissions to show a dynamic name
+    const { write } = embeddable.isReadOnlyEnabled();
+    return write
+      ? i18n.translate('presentationPanel.action.editPanel.displayName', {
+          defaultMessage: 'Edit {value}',
```

Comment:
I am a little confused as to why this will show `Edit` while dashboard is in view mode. Maybe I missed part of the conversation?

Reply:
From @MichaelMarcialis review here: https://github.com/elastic/kibana/pull/208554/files/34ff73b5fdab54a0e798ddbdd09203b7e9401ee8#r1936137458

He proposes to change it to `Edit configuration` while I'm taking one step at the time here to the current `Edit` label and rework in a follow up both this `Edit` and `Show` action labels.

Reply:
But instead of showing and `edit` label, why not just show the edit action? I guess that's why I am confused. What is the difference between show action with write capabilities and existing edit action? I guess this just really muddies the water between the 2.

Reply:
In case it helps, I've added [updated guidance above](https://github.com/elastic/kibana/pull/208554#discussion_r1939693883) based on the latest changes and screenshots. Let me know if this makes sense or if there are additional questions/concerns.